br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -564,6 +577,10 @@ func (m *DupeDetector) RecordIndexConflictError(ctx context.Context, stream DupK
 			}
 			indexHandles.truncate()
 		}
+
+		if algorithm == config.DupeResAlgErr {
+			return NewErrFoundConflictRecords(ctx, key, val, m.tbl, indexInfo)
```

Comment:
We can extract the part of NewErrFoundConflictRecords into ErrFoundIndexConflictRecords. 
NewErrFoundConflictRecords will check if the key is a record key and construct a ErrFoundDataConflictRecords, otherwise find the indexInfo from tableInfo and call ErrFoundIndexConflictRecords.
And here we can directly call ErrFoundIndexConflictRecords.